src/pages/register-a-beacon/check-beacon-summary.tsx

```diff
@@ -0,0 +1,139 @@
+            </>
+          }
+        />
+      </Layout>
+    </>
+  );
+};
+
+const BeaconSummary: FunctionComponent<BeaconDetailsProps> = ({
+  beaconManufacturer,
+  beaconModel,
+  beaconHexId,
+}: BeaconDetailsProps): JSX.Element => (
+  <SummaryList>
+    <SummaryListItem
+      labelText="Beacon manufacturer"
+      valueText={beaconManufacturer}
+    />
+    <SummaryListItem labelText="Beacon model" valueText={beaconModel} />
+    <SummaryListItem labelText="Beacon HEX ID" valueText={beaconHexId} />
+    <SummaryListItem
+      labelText="Date registered"
+      // TODO: Lookup for date registered if beacon already in system
+      valueText="19 September 2016"
+    />
+  </SummaryList>
+);
+
+const BeaconAlreadyRegistered: FunctionComponent = () => (
+  <NotificationBanner title="This beacon is already registered">
+    <div className="govuk-body">
+      There maybe a few reasons why this beacon is already registered,
+      including:
+    </div>
+    <ul className="govuk-list govuk-list--bullet">
+      <li>
+        It has already been registered by yourself or someone else (e.g. if you
+        bought it second hand)
+      </li>
+      <li>The manufacturer may have incorrectly reused the HEX ID</li>
+    </ul>
+  </NotificationBanner>
+);
+
+const WhatNext: FunctionComponent = () => (
+  <Form action="/">
+    <FormGroup>
+      <FormFieldset>
+        <FormLegend>What would you like to do next?</FormLegend>
+        <RadioList>
+          <RadioListItem
+            id="try-again"
+            name="do-next"
+            value="try-again"
+            text="Try again or register a different beacon"
+          />
+          <RadioListItemHint
+            id="continue-anyway"
+            name="do-next"
+            value="continue-anyway"
+            text="Continue anyway - I've entered the beacons details correctly. I think this is a manufacturer fault or want to update the owner details"
+            hintText="We will contact the beacon manufacturer to rectify this. In the meantime, you can still use your beacon and Search & Rescue will still be able to locate you."
+          />
+        </RadioList>
+      </FormFieldset>
+    </FormGroup>
+  </Form>
+);
+
+// TODO: Encapsulate the state caching function
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  if (context.req.method === "POST") {
+    // TODO: Investigate more widely used library for parse()
+    const previousFormPageData: BeaconDetailsProps = await parse(context.req);
+
+    const state: IFormCache = FormCacheFactory.getState();
+    state.set("id", previousFormPageData);
+
+    return {
+      props: previousFormPageData,
+    };
+  } else if (context.req.method === "GET") {
+    const state: IFormCache = FormCacheFactory.getState();
+    const existingState = state.get("id");
+
+    return { props: existingState };
+  }
+
+  return {
+    props: {},
+  };
+};
```

Comment:
[getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering) is called by NextJS on first rendering of the page.  It sets the props that are injected into the page.

The `context` argument contains the request that prompted the page load.  

We've set it up here so the previous page in the form flow (`check-beacon-details`) sends a `POST` request to this page, which is then extracted in `getServerSideProps` and added to the [form state cache](https://github.com/madetech/mca-beacons-webapp/blob/51dfdc7683ff8264eb737f47d0754828c7d8c845/src/lib/form-cache.ts).

Other branch of the conditional handles normal `GET` requests not originating from a form submission.